rapids_logger/CMakeLists.txt

```diff
@@ -0,0 +1,195 @@
+  # paths incorrectly for configure_file/install.
+  set(BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR}/${_RAPIDS_LOGGER_HEADER_DIR})
+  set(INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/${_RAPIDS_LOGGER_HEADER_DIR})
+
+  set(LOGGER_OUTPUT_FILE ${BUILD_DIR}/logger.hpp)
+  configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger.hpp.in ${LOGGER_OUTPUT_FILE})
+  install(FILES ${LOGGER_OUTPUT_FILE} DESTINATION ${INSTALL_DIR})
+
+  set(LOGGER_IMPL_OUTPUT_FILE ${BUILD_DIR}/logger_impl/logger_impl.hpp)
+  configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger_impl.hpp.in ${LOGGER_IMPL_OUTPUT_FILE})
+  install(FILES ${LOGGER_IMPL_OUTPUT_FILE} DESTINATION ${INSTALL_DIR}/logger_impl)
+
+  add_library(${_RAPIDS_LOGGER_TARGET} INTERFACE)
+  include(GNUInstallDirs)
+  # Note: The BUILD_INTERFACE setting assumes that LOGGER_HEADER_DIR is the subdirectory of
+  # CMAKE_INSTALL_INCLUDEDIR relative to which all includes are rooted in the C++ code files. I
+  # think that is a safe assumption though since if it were violated then the INSTALL_INTERFACE
+  # would not only be incorrect (if computed using LOGGER_HEADER_DIR), but it would also break
+  # consumers of the installed package who expect to be able to write `#include
+  # <${LOGGER_HEADER_DIR/include\//}/logger.hpp>` and have it work.
+  target_include_directories(
+    ${_RAPIDS_LOGGER_TARGET}
+    INTERFACE "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}>"
+              "$<INSTALL_INTERFACE:${_RAPIDS_LOGGER_HEADER_DIR}>")
+  target_compile_features(${_RAPIDS_LOGGER_TARGET} INTERFACE cxx_std_17)
+  if(_RAPIDS_SUPPORTS_LOGGING)
+    target_compile_definitions(${_RAPIDS_LOGGER_TARGET}
+                               INTERFACE "${_RAPIDS_LOGGER_MACRO_PREFIX}_SUPPORTS_LOGGING")
+
+    # Create an interface target that will trigger compilation of the logger implementation in any
+    # target that is linked to it.
+    set(LOGGER_IMPL_SRC_OUTPUT_FILE ${BUILD_DIR}/logger_impl/logger.cpp)
+    configure_file(${CMAKE_CURRENT_FUNCTION_LIST_DIR}/logger.cpp.in ${LOGGER_IMPL_SRC_OUTPUT_FILE})
+    install(FILES ${LOGGER_IMPL_SRC_OUTPUT_FILE} DESTINATION ${INSTALL_DIR}/logger_impl)
+
+    # Note that we cannot specify the source files directly in add_library, see the CMake
+    # documentation explaining that these do not populate INTERFACE_SOURCES.
+    # https://cmake.org/cmake/help/latest/command/add_library.html#interface-with-sources
+    set(impl_target ${_RAPIDS_LOGGER_TARGET}_impl)
+    add_library(${impl_target} INTERFACE)
+    target_sources(
+      ${impl_target}
+      INTERFACE $<BUILD_INTERFACE:${LOGGER_IMPL_SRC_OUTPUT_FILE}>
+                $<INSTALL_INTERFACE:${_RAPIDS_LOGGER_HEADER_DIR}/logger_impl/logger.cpp>)
+    target_link_libraries(${impl_target} INTERFACE ${_RAPIDS_LOGGER_TARGET} spdlog::spdlog)
+    # TODO: Figure out whether to use this or the visibility settings at compile-time or both.
+    target_link_options(${impl_target} INTERFACE "LINKER:--exclude-libs,libspdlog")
```

Comment:
There are very subtle differences in which symbols leak based on using one way or the other of symbol hiding. In the grand scheme of things I suspect that all these edge case symbols are harmless to allow leakage of, but I would like some confirmation. @robertmaynard when you have a few moments could we take a look and confirm?

Reply:
Removing for now, will add back when we switch to static linking.